lib/dependencies/HarmonyTopLevelThisParserPlugin.js

```diff
@@ -0,0 +1,26 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Florent Cailhol @ooflorent
+*/
+"use strict";
+
+const ConstDependency = require("./ConstDependency");
+
+class HarmonyTopLevelThisParserPlugin {
+	apply(parser) {
+		let isHarmony = false;
```

Comment:
Is there a separate instance of `(new HarmonyTopLevelThisParserPlugin()).apply` for each module, or is this taking advantage of a race condition?

Reply:
> is this taking advantage of a race condition?

Yes. It reads metadata set by `HarmonyDetectionParserPlugin`.